netaddr.go

```diff
@@ -383,27 +317,18 @@ func (ip IP) IPAddr() *net.IPAddr {
 //
 // It returns false for IP4-mapped IPv6 addresses. See IP.Unmap.
 func (ip IP) Is4() bool {
-	if ip.ipImpl == nil {
-		return false
-	}
-	return ip.ipImpl.is4()
+	return ip.zt == z4
 }
 
 // Is4in6 reports whether ip is an IPv4-mapped IPv6 address.
 func (ip IP) Is4in6() bool {
-	if ip.ipImpl == nil {
-		return false
-	}
-	return ip.ipImpl.is4in6()
+	return ip.zt == z6noz && string(ip.a[:len(mapped4Prefix)]) == mapped4Prefix
```

Comment:
Should v6+zone return true as well?

Reply:
We dug into this separately, and turns out v4 link-local addresses are supposed to be used by binding to the specific interface you want to send out of, rather than using a zone specifier. Yay incompatible APIs!